### ISSUE TITLE
Allow edits to previous years unit submissions

### ISSUE THREAD

**Issue by anterick-cjx** [vEnhance/otis-web#318]
Basically I tried to "submit additional problems" for last year's unit since otherwise I'd have to request it again and that takes too long, but it says you can't do that because the semester isn't active. I think it'd be a lot more efficient if we were allowed to submit from previous years or something like that.

**Comment by Incompleteusern** [comment]
I think the consensus is that https://github.com/vEnhance/otis-web/issues/300 is the easiest way of implementing this.

**Comment by vEnhance** [comment]
I agree that finishing #300 (i.e. allowing automatic approval of requests of units that were unlocked in a previous year) would provide a workaround here that isn't too annoying.

It's more complicated to try and resubmit to an inactive semester, because the pset/unit logic is based on ordered pairs of (human, year) rather than just the human. So, e.g. if a student submits a unit in year Y and then resubmits it in year Y+1 it's no longer clear which year that submission should be attached to.

In the long term maybe we should just drop the year dependence; https://github.com/vEnhance/otis-web/commit/1ebab2a6 was a related issue caused by the extra dependence. However, that's quite long-term because it requires refactoring a ton of code.
